.ci/update-tekton-bundle.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+TEMP_FOLDER=$WORKSPACE/tmp/bundle-override
+APPSTUDIO_QE_REPO=quay.io/redhat-appstudio-qe/test-images
+TASK_BUNDLE_IMG=$APPSTUDIO_QE_REPO:task-bundle-$JVM_BUILD_SERVICE_PR_SHA
+export PIPELINE_BUNDLE_IMG=$APPSTUDIO_QE_REPO:pipeline-bundle-$JVM_BUILD_SERVICE_PR_SHA
+
+function setTargetBundle() {
+    TARGET_BUNDLE=$(yq e ".configMapGenerator[].literals[]" \
+    "$WORKSPACE"/tmp/infra-deployments/components/build/kustomization.yaml \
+    | grep default_build_bundle \
+    | sed "s/default_build_bundle=//")
+
+    echo "New build bundle will be created based on $TARGET_BUNDLE"
+}
+
+function createPipelinesFile() {
+    pipelines=$(tkn bundle list "$TARGET_BUNDLE" | sed "s/pipeline.tekton.dev\///")
+    for pipeline in $pipelines; do
+        echo "---" >> "$TEMP_FOLDER"/pipelines.yaml
+        tkn bundle list -o yaml "$TARGET_BUNDLE" pipeline "$pipeline" >> "$TEMP_FOLDER"/pipelines.yaml
+    done
+}
+
+function createTaskFile() {
+    for bundle in $(grep bundle "$TEMP_FOLDER"/pipelines.yaml | sort -u | awk '{print $2}'); do 
+        tasks=$(tkn bundle list "$bundle" | sed "s/task.tekton.dev\///")
+
+        for task in $tasks; do
+            if [ "$task" == "s2i-java" ]; then
+                tkn bundle list -o yaml "$bundle" task "$task" >> "$TEMP_FOLDER"/task.yaml
+            fi
+        done
+    done
+}
+
+function updateAnalyzerImage() {
+    echo "jvm-build-service analyzer image set to $JVM_BUILD_SERVICE_ANALYZER_IMAGE"
+    yq e -i "select(.metadata.name == \"s2i-java\") \
+    | (.spec.steps[] | select(.name == \"analyse-dependencies-java-sbom\").image) \
+    |= \"$JVM_BUILD_SERVICE_ANALYZER_IMAGE\"" "$TEMP_FOLDER"/task.yaml
+}
+
+function updatePipelineRef() {
+    yq e -i "(.spec.tasks[].taskRef | select (.name == \"s2i-java\").bundle) \
+    |= \"$TASK_BUNDLE_IMG\"" "$TEMP_FOLDER"/pipelines.yaml
+}
+
+function createDockerConfig() {
+    mkdir ~/.docker
+    echo "$QUAY_TOKEN" | base64 -d > ~/.docker/config.json
+}
+
+function createAndPushTaskBundle() {
+    echo "Creating $TASK_BUNDLE_IMG"
+    tkn bundle push "$TASK_BUNDLE_IMG" -f "$TEMP_FOLDER"/task.yaml
+}
+
+function createAndPushPipelineBundle() {
+    echo "Creating $PIPELINE_BUNDLE_IMG"
+    tkn bundle push "$PIPELINE_BUNDLE_IMG" -f "$TEMP_FOLDER"/pipelines.yaml
+}
+
+mkdir "$TEMP_FOLDER"
+
+setTargetBundle
+createPipelinesFile
+createTaskFile
+updateAnalyzerImage
+updatePipelineRef
+createDockerConfig
+createAndPushTaskBundle
+createAndPushPipelineBundle
```

Comment:
how is new Pipeline bundle used?

Reply:
Short term, I do not believe it is, because of the "hard coding" we sorted out with https://coreos.slack.com/archives/C032EJ007C0/p1658212125508779

The CI flow will set `JVM_BUILD_SERVICE_REQPROCESSOR_IMAGE` to the image built in openshift CI, and then because of https://github.com/redhat-appstudio/infra-deployments/blob/main/components/build/jvm-build-service/operator-images.yaml#L12-L13 the controller will use that openshift CI built image when it in-lines the req analyzer task per https://github.com/redhat-appstudio/jvm-build-service/blob/main/pkg/reconciler/artifactbuild/artifactbuild.go#L461 and https://github.com/redhat-appstudio/jvm-build-service/blob/main/pkg/reconciler/dependencybuild/dependencybuild.go#L508.

But longer term, as we switch to pipeline runs, and as we make progress on https://docs.google.com/document/d/1M4o4cMMYz8lM_gd4VfJGiLE8p9FBWeeQvOxcfq9WYDE/edit# minimally (and maybe on other stories?), and presumably build-service can decipher that an incoming Component needs `jvm-build-service` integration, then it will be access the bundle that is being created here, per the configmaps based overrides that @brunoapimentel added a month or 2 ago, as it constructs the PIpelineRun needed for the build, and away we go.

So rather than trying to wedge in CI / e2e updates after the fact, yes, this PR establishes the test plumbing before we actually implement features on the horizon.

Reply:
and yes I now (re?)realize this change is not an immediate blocker to PLNSRVCE-452

Reply:
I'm not sure I follow everything you wrote there, @gabemontero. :sweat_smile: 

I understood that the e2e tests would create a configmap that points to the new bundle as needed, and that's how the pipeline would be pulled in.

We discussed it on this thread: https://github.com/redhat-appstudio/jvm-build-service/pull/103#discussion_r926277674

Reply:
yes that still holds @brunoapimentel .... what I clarified in https://github.com/redhat-appstudio/jvm-build-service/pull/103#discussion_r928909364 is my realization on _**WHEN**_ that confirmap would actually have an effect.  When the _**WHEN**_ is after we stop hardcoding those task specs in the controller and changes them to refs that point to items in the bundle.

Reply:
and yes the wiring of all this is crazy complicated ... I myself may still not have the connections 100% memorized and accurately "recall-able".